bpf/lib/nodeport.h

```diff
@@ -125,9 +131,14 @@ nodeport_add_tunnel_encap_opt(struct __ctx_buff *ctx, __u32 src_ip, __be16 src_p
 			return ret;
 	}
 
-	return __encap_with_nodeid_opt(ctx, src_ip, src_port, dst_ip,
-				       src_sec_identity, dst_sec_identity, NOT_VTEP_DST,
-				       opt, opt_len, ct_reason, monitor, ifindex);
+	if (info->flag_ipv6_tunnel_ep)
+		return __encap_with_nodeid_opt6(ctx, &info->tunnel_endpoint.ip6,
+						src_sec_identity, info->sec_identity,
```

Comment:
```suggestion
						src_sec_identity, dst_sec_identity,
```

until we're confident that all callers pass the identical identity in the `remote_endpoint_info`, and can then also use it in the IPv4 path?

Reply:
I went the other way around and completely remove the `dst_sec_identity` argument.